api/tests/test_views.py

```diff
@@ -129,17 +130,70 @@ def test_basic_list(self):
 
     def test_auth_list(self):
         """An authenticated user should get more SMS media back."""
-        unauth_response_data = self.view(self.factory.get('/')).data
+        unauth_response_data = self.view(self.get_request).data
 
-        request = self.factory.get('/')
-        force_authenticate(request, user=self.user)
-        auth_response_data = self.view(request).data
+        force_authenticate(self.get_request, user=self.user)
+        auth_response_data = self.view(self.get_request).data
 
         # Authorised users have more results
         self.assertGreater(
             len(auth_response_data['results']), len(unauth_response_data['results']))
 
 
+class MediaViewTestCase(ViewTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.view = views.MediaView().as_view()
+
+    @mock.patch('smsjwplatform.jwplatform.DeliveryVideo.from_key')
+    def test_success(self, mock_from_id):
+        """Check that a media item is successfully returned"""
+        mock_from_id.return_value = api.DeliveryVideo(DELIVERY_VIDEO_FIXTURE)
+
+        # test
+        response = self.view(self.get_request, 'XYZ123')
+
+        mock_from_id.assert_called_with('XYZ123')
+
+        self.assertEqual(response.status_code, 200)
+
+        self.assertEqual(response.data['id'], 'mock1')
+        self.assertEqual(response.data['title'], 'Mock 1')
+        self.assertEqual(response.data['description'], 'Description for mock 1')
+        self.assertEqual(response.data['published_at_timestamp'], 1234567)
+        self.assertEqual(
+            response.data['poster_image_url'], 'https://cdn.jwplayer.com/thumbs/mock1-720.jpg'
+        )
+        self.assertEqual(response.data['duration'], 54.0)
+        self.assertEqual(response.data['ui_url'], '/media/mock1')
```

Comment:
Can we test against what Django's ``reverse`` helper returns? Then is we change the URL, we don't need to change all of the tests.